plugins/modules/aws_s3_bucket_info.py

```diff
@@ -57,50 +182,223 @@
     pass  # Handled by AnsibleAWSModule
 
 from ansible.module_utils._text import to_native
-
 from ansible_collections.amazon.aws.plugins.module_utils.core import AnsibleAWSModule
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import AWSRetry, boto3_tag_list_to_ansible_dict
 from ansible_collections.amazon.aws.plugins.module_utils.ec2 import camel_dict_to_snake_dict
 
 
-def get_bucket_list(module, connection):
+def get_bucket_list(module, connection, name="", name_filter=""):
     """
     Return result of list_buckets json encoded
+    Filter only buckets matching 'name' or name_filter if defined
     :param module:
     :param connection:
     :return:
     """
+    buckets = []
+    filtered_buckets = []
+    final_buckets = []
+
+    # Get all buckets
     try:
         buckets = camel_dict_to_snake_dict(connection.list_buckets())['buckets']
-    except botocore.exceptions.ClientError as e:
-        module.fail_json(msg=to_native(e), exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+    except botocore.exceptions.ClientError as err_code:
+        module.fail_json(msg=to_native(err_code), exception=traceback.format_exc(), **camel_dict_to_snake_dict(err_code.response))
```

Comment:
```suggestion
        module.fail_json_aws(err, msg="Failed to fetch bucket list")
```

We have a nice helper which knows how to cleanly display both types of Boto related errors and gives you the opportunity to also explain what you were attempting to do at the time.  You'll see a lot of examples which catch BotoCore- and Client- Errors separately because BotoCore doesn't have a response item so had to be handled as a special case.  These generally pre-date some of our helpers.  Unfortunately cleaning up results in significantly sized PRs (#361) which are a PITA to review.